components/prism-wgsl.js

```diff
@@ -0,0 +1,67 @@
+Prism.languages.wgsl = {
+    'comment': {
+        pattern: /\/\/.*|\/\*[\s\S]*?(?:\*\/|$)/,
+        greedy: true,
+    },
+    'attributes': {
+        pattern: /(@)(?:align|binding|builtin|const|group|id|interpolate|invariant|location|size|workgroup_size|vertex|fragment|compute)/i,
+        lookbehind: true,
+        alias: 'attr-name',
+    },
+    'functions': {
+        pattern: /\b(fn\s+)[_a-zA-Z]\w*(?=[(<])/,
+        lookbehind: true,
+        alias: 'function',
+    },
+    'keyword': /\b(?:bitcast|break|case|const|continue|continuing|default|discard|else|enable|fallthrough|fn|for|function|if|let|loop|private|return|storage|switch|type|uniform|var|while|workgroup)\b/,
+    'function-calls': {
+        pattern: /\b[_a-z]\w*(?=\()/i,
+        alias: 'function',
+    },
+    'builtin': /\b(?:array|atomic|bool|i32|u32|f32|i64|u64|f64|vec[2-4]|mat[2-4]x[2-4]|override|ptr|sampler|sampler_comparison|staticAssert|struct|texture_1d|texture_2d|texture_2d_array|texture_3d|texture_cube|texture_cube_array|texture_multisampled_2d|texture_storage_1d|texture_storage_2d|texture_storage_2d_array|texture_storage_3d|texture_depth_2d|texture_depth_2d_array|texture_depth_cube|texture_depth_cube_array|texture_depth_multisampled_2d)\b/,
+    'class-name': /\b(?:[A-Z][A-Za-z0-9]*)\b/,
+    'built-in-values-vertex': {
+        pattern: /\b(?:vertex_index|instance_index|position)\b/,
+        alias: 'attr-value',
+    },
+    'built-in-values-fragment': {
+        pattern: /\b(?:front_facing|frag_depth|sample_index|sample_mask)\b/,
+        alias: 'attr-value',
+    },
+    'built-in-values-compute': {
+        pattern: /\b(?:local_invocation_id|local_invocation_index|global_invocation_id|workgroup_id|num_workgroups)\b/,
+        alias: 'attr-value',
```

Comment:
I'm getting false positives for these.

![image](https://user-images.githubusercontent.com/20878432/168777998-d7491b23-6b2b-4bc8-b5de-dca427bb147f.png)
![image](https://user-images.githubusercontent.com/20878432/168778041-26387244-38d6-476f-a5e8-7e7eed08e400.png)

I think this could be fixed by saying that built-ins must not be followed by a `:` or preceded by a `.`.
```suggestion
    'built-in-values-vertex': {
        pattern: /((?:^|[^.\s])\s*)\b(?:vertex_index|instance_index|position)\b(?!\s*:)/,
        lookbehind: true,
        alias: 'attr-value',
    },
    'built-in-values-fragment': {
        pattern: /((?:^|[^.\s])\s*)\b(?:front_facing|frag_depth|sample_index|sample_mask)\b(?!\s*:)/,
        lookbehind: true,
        alias: 'attr-value',
    },
    'built-in-values-compute': {
        pattern: /((?:^|[^.\s])\s*)\b(?:local_invocation_id|local_invocation_index|global_invocation_id|workgroup_id|num_workgroups)\b(?!\s*:)/,
        lookbehind: true,
        alias: 'attr-value',
```

Reply:
Another options might be to only match them only inside the 'builtin' attribute, as they are only defined in that context. I think you can actually name a variable or a property 'position' but i havent tried it. Is this possible with inside?

Reply:
Yes, that would be possible using `inside` for `attribute`.

Reply:
Ok i did it with `inside` now, but now in `builtin(notABuiltInValue)` the `notABuiltInValue` gets highlighted as attribute. Can i somehow achieve that everything inside the `inside` that doesn't get matched will not be highlighted as attribute. Should i use some kind of catch all pattern or is there another way to do this?

Reply:
We could add a helper token for this.

```js
'builtin-attribute': {
  pattern: /(@)builtin\(.*?\)/,
  lookbehind: true,
 inside: {
    'attribute': /^builtin/, // TODO: alias
    'punctuation': /[(),]/,
    'built-in-values': /\b(?:...)\b/ // TODO: alias
  }
},    
'attributes': {
        pattern: /(@)(?:align|binding|const|group|id|interpolate|invariant|location|size|workgroup_size|vertex|fragment|compute)/i,
        lookbehind: true,
        alias: 'attr-name'
}
```